tests/generic_config_updater/test_ipv6.py

```diff
@@ -0,0 +1,196 @@
+import logging
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success, expect_op_failure
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+from tests.generic_config_updater.gu_utils import create_checkpoint, delete_checkpoint, rollback_or_reload
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.sanity_check(skip_sanity=True)
+]
+
+@pytest.fixture(autouse=True)
+def ensure_dut_readiness(duthost):
+    """
+    Setup/teardown fixture for each ipv6 test
+    rollback to check if it goes back to starting config
+
+    Args:
+        duthost: DUT host object under test
+    """
+
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+    finally:
+        delete_checkpoint(duthost)
+
+
+def get_ipv6_neighbor(duthost):
+    """
+    Returns ipv6 BGP neighbor address, properties of BGP neighbor
+
+    Args:
+        duthost: DUT host object
+    """
+
+    config_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
+    bgp_neighbors_data = config_facts['BGP_NEIGHBOR']
+    for neighbor_address in bgp_neighbors_data.keys():
+        if "::" in neighbor_address:
```

Comment:
>"::" [](http://example.com/codeflow?start=11&length=4)

To check ipv6 address, the stricter way is to check the type of `ipaddress.ip_address(neighbor_address)`.